content/blog/recipes-0-2-0/index.Rmd

```diff
@@ -0,0 +1,86 @@
+title: recipes 0.2.0
+date: 2022-02-25
+author: Max Kuhn
+description: >
+    Recipes has added a few new steps along with many improvements.
+
+photo:
+  url: https://unsplash.com/photos/xZa4JUE7EdM
+  author: Jonny Clow
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [package] 
+tags: [recipes tidymodels]
+---
+
+<!--
+TODO:
+* [ ] Look over / edit the post's title in the yaml
+* [ ] Edit (or delete) the description; note this appears in the Twitter card
+* [ ] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [ ] Find photo & update yaml metadata
+* [ ] Create `thumbnail-sq.jpg`; height and width should be equal
+* [ ] Create `thumbnail-wd.jpg`; width should be >5x height
+* [ ] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+We're very excited to announce the release of [recipes](https://recipes.tidymodels.org/) 0.2.0. recipes is a package for preprocessing data before using it in models or visualizations. You can think of it as a machup of `model.matrix()` and dplyr. 
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("recipes")
+```
+
+This blog post will describe what's new. You can see a full list of changes in the [release notes](https://github.com/tidymodels/recipes/blob/main/NEWS.md)
+
+## New Steps
+
+`step_nnmf_sparse()` was added to produce features using non-negative matrix factorization (via the [RcppML](https://github.com/zdebruine/RcppML) package). This will supersede the existing `step_nnmf()` since the package requirements for that step were difficult to support and used. The new step allows for a sparse representation via regularization and, from our initial testing, to be **much faster** than the original NNMF step. 
+
+A new function helps create indicator variables from text data, especially those with multiple choice values. `step_dummy_extract()`. For example, if a row of a variable had a value of `"red,black,brown"`, the step can separate these values and make all of the required binary dummy variables. 
+
+Here's a real example from [Episode 8 of _Sliced_](https://www.kaggle.com/c/sliced-s01e08-KJSEks) where a column of data from Spotify had the artist(s) of a song: 
+
+```{r}
+library(recipes)
+spotify <- 
+  tibble::tribble(
+    ~ artists,
+    "['Genesis']",
+    "['Billie Holiday', 'Teddy Wilson']",
+    "['Jimmy Barnes', 'INXS']"
+  )
+recipe(~ artists, data = spotify) %>% 
+  step_dummy_extract(artists, pattern = "(?<=')[^',]+(?=')") %>% 
+  prep() %>% 
+  bake(new_data = NULL) %>% 
+  names()
+```
+
+Note that this step produces an "other" column and has arguments similar to `step_other()`. 
+
+`step_percentile()`, previously seem in the developer documentation, can determine the empirical distribution of the variable using the training set, then convert any value to the percentile of this distribution. 
```

Comment:
```suggestion
`step_percentile()` is a new step function after it had previously only been an example in the developer documentation. It can determine the empirical distribution of a variable using the training set, then convert any value to the percentile of this distribution. 
```